### PR TITLE
Data Explorer - Reverse years and fix firstColumn ordering in table

### PR DESCRIPTION
**Pull request by Bluesmile82**
This PR orders the column name years from newer to older in the Data Explorer

image

### CODE REVIEW

File: app/javascript/app/components/table/table-component.jsx [hunk 1]
```diff
@@ -58,9 +58,11 @@ class SimpleTable extends PureComponent {
     if (!data.length) return null;
     const hasColumnSelectedOptions = hasColumnSelect && columnsOptions;
     const activeColumnNames = activeColumns.map(c => c.value);
-    const columnData = activeColumnNames
-      .filter(c => firstColumnHeaders.includes(c))
-      .concat(difference(activeColumnNames, firstColumnHeaders));
+    let columnData = [];
```
Comment (aabdaab) [inline app/javascript/app/components/table/table-component.jsx]: what was wrong with `filter` function?

Comment (Bluesmile82) [reply app/javascript/app/components/table/table-component.jsx]: It was just adding the firstColumns first. Not in the same order that they were passed. This is a quite sensible prop as I want to make it relatively generic

Comment (aabdaab) [reply app/javascript/app/components/table/table-component.jsx]: huh, I thought `filter` was preserving the order. ok then 👍

**Comment by simaob** [comment]
I've merged another PR and now there are conflicts... sorry. Please check them out!

File: app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js [hunk 2]
```diff
@@ -542,6 +543,15 @@ export const parseData = createSelector([getData], data => {
   return updatedData.map(d => pick(d, whiteList));
 });
 
+export const getFirstColumnHeaders = createSelector([parseData], data => {
+  if (!data || !data.length) return null;
+  const isANumber = i => !isNaN(parseInt(i, 10));
+  const yearColumnKeys = Object.keys(data[0])
+    .filter(k => isANumber(k))
+    .reverse();
+  return DATA_EXPLORER_FIRST_COLUMN_HEADERS.concat(yearColumnKeys);
```
Comment (weberjavi) [inline app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js]: Wouldn't it be more accurate to name this as `DATA_EXPLORER_TABLE_HEADERS`? The `first_column` bit could be misleading?

File: app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js [hunk 3]
```diff
@@ -542,6 +543,15 @@ export const parseData = createSelector([getData], data => {
   return updatedData.map(d => pick(d, whiteList));
 });
 
+export const getFirstColumnHeaders = createSelector([parseData], data => {
```
Comment (weberjavi) [inline app/javascript/app/components/data-explorer-content/data-explorer-content-selectors.js]: Could this be named something like `getHeadersWithYears` as it describes better what's happening inside the function?

**Comment by simaob** [comment]
@Bluesmile82 did you see Javi's comments, any further action from those?

**Comment by weberjavi** [comment]
Merging

**Pull request merged**